operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java

```diff
@@ -122,6 +137,41 @@ private static void calculateNamespace() {
     namespace = "keycloak-test-" + UUID.randomUUID();
   }
 
+  protected static void deployDB() {
+    // DB
+    Log.info("Creating new PostgreSQL deployment");
+    k8sclient.load(KeycloakDeploymentE2EIT.class.getResourceAsStream("/example-postgres.yaml")).inNamespace(namespace).createOrReplace();
+
+    // Check DB has deployed and ready
+    Log.info("Checking Postgres is running");
+    Awaitility.await()
+            .untilAsserted(() -> assertThat(k8sclient.apps().statefulSets().inNamespace(namespace).withName("postgresql-db").get().getStatus().getReadyReplicas()).isEqualTo(1));
+  }
+
+  // TODO improve this (preferably move to JOSDK)
+  protected void savePodLogs() {
+    Log.infof("Saving pod logs to %s", POD_LOGS_DIR);
+    for (var pod : k8sclient.pods().inNamespace(namespace).list().getItems()) {
+      try {
+        String podName = pod.getMetadata().getName();
+        Log.infof("Processing %s", podName);
+        String podLog = k8sclient.pods().inNamespace(namespace).withName(podName).getLog();
+        File file = new File(POD_LOGS_DIR + String.format("%s-%s.txt", namespace, podName)); // using namespace for now, if more tests fail, the log might get overwritten
+        file.getAbsoluteFile().getParentFile().mkdirs();
+        try (var fw = new FileWriter(file, false)) {
+          fw.write(podLog);
```

Comment:
what would be the point of this on CI ? Can we review these files on a failed pipeline ?

Reply:
Created #10140.